### ISSUE TITLE
sea surface temperature 

### ISSUE THREAD

**Issue by LH002** [acolite/acolite#15]
Hello, can acolite retrieve sea surface temperature?

**Comment by acolite** [comment]
Yes, through the use of TACT. This can be activated by setting tact_run=True in your settings file. Note that you need to have libRadtran and UCAR RDA access as described in the https://github.com/acolite/acolite/blob/main/README.md file and the user manual https://github.com/acolite/acolite/releases/download/20210802.0/acolite_manual_20210802.0.pdf

Let me know how it goes!

Quinten

**Comment by acolite** [comment]
If you are comfortable using Docker, I have provided a Docker image that supports TACT processing: https://odnature.naturalsciences.be/remsem/acolite-forum/viewtopic.php?p=1235#p1235

Quinten

**Comment by LH002** [comment]
&nbsp;Thank you very much. I will try it.



---Original---
From: "Quinten ***@***.***&gt;
Date: Mon, Nov 22, 2021 23:03 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [acolite/acolite] sea surface temperature  (Issue #15)




 
If you are comfortable using Docker, I have provided a Docker image that supports TACT processing: https://odnature.naturalsciences.be/remsem/acolite-forum/viewtopic.php?p=1235#p1235
 
Quinten
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by LH002** [comment]
Dear Quinten:
    I encountered an error while installing libradtran. The error occurred when I ran Mack check. The specific error information is as follows:

MYSTIC spherical (0.00001,   2.0)..............................................sh: 1: gawk: not found

Different number of lines in files

  ../examples/UVSPEC_MC_SPHERICAL.OUT            3

  TMP.OUT           -1

Aborting

 at ../src/ndiff line 44.

ndiff failed at test.pl line 1907.

	main::report_differences(0, "TMP.OUT", "../examples/UVSPEC_MC_SPHERICAL.OUT", "TMP.INP", "MYSTIC spherical", "", 0.02, 1e-05) called at test.pl line 1616

	main::mystic_run_spc_test("TMP.INP", "TMP.OUT", "../examples/UVSPEC_MC_SPHERICAL.INP", "../examples/UVSPEC_MC_SPHERICAL.OUT", "MYSTIC spherical", 0, 0.02, 1e-05) called at test.pl line 889

7 of the uvspec tests failed.

make[1]: *** [Makefile:17: all] Error 1

make[1]: Leaving directory '/home/lh/acolite-main/external/libRadtran-2.0.2/test'

make: *** [Makefile:63: check] Error 2

How can I solve this problem?

**Comment by acolite** [comment]
Do you have gawk installed?

**Comment by LH002** [comment]
Gawk was not installed on my Linux system before, but it is now installed. After the installation, the error message appears again. The error message is as follows:
MYSTIC spectral importance sampling (REPTRAN) (0.00001,   5.0).................    2 serious differences.  Maximum difference:   5.420000%. Absolute values (test,ans): (2.954151e-04, 2.793972e-04).

MYSTIC boxairmass factor (0.00001,   1.0)...................................... ok.

MYSTIC spherical (0.00001,   2.0).............................................. ok.

2 of the uvspec tests failed.

make[1]: Leaving directory '/home/lh/acolite-main/external/libRadtran-2.0.2/test'

What should I do now？

**Comment by acolite** [comment]
It seems like the compilation went fine, and most tests passed. The message you get is because the values in one of the tests exceeds the set threshold. (The values differ by just over 5%.) Since MYSTIC is a Monte Carlo solver this is probably OK. If you have questions about libRadtran, the solvers, or the meaning of these tests, you should get in touch with the developers!

You should be all set up to use TACT now!

Quinten

**Comment by LH002** [comment]
Thank you！

**Comment by LH002** [comment]
Dear Quinten:
I ran and processed the function of TACT, and the results are as follows:
![1652060095(1)](https://user-images.githubusercontent.com/35960535/167326123-51ec1d19-df84-46ee-b3ce-27260b280f1f.png)
Does my TACT function handle normally? Which product is the final SST?

**Comment by acolite** [comment]
The ST NetCDF contains the TACT outputs. Surface Temperature products are given by st10 and  st11 (respectively in bands 10 and 11). Note that you need to apply a mask to land if you just want the SST data.

The parameters in the ST.nc file are (for each band):
- bt at-sensor brightness temperature
- em emissivity (defaults to water emissibity)
- Ld downwelling atmospheric radiance
- ls surface emitted radiance
- lt total at-sensor radiance
- Lu upwelling atmospheric radiance
- st surface temperature
- tau atmospheric transmittance

For more details I would refer to: https://doi.org/10.1016/j.rse.2019.111518

Quinten

**Comment by LH002** [comment]
Thank you！

**Comment by jhonnatan2178** [comment]
Hi, were you able to apply the mask to the land? I am trying to retrieve SST, but also checking for temperature in some rivers mouth

**Comment by acolite** [comment]
You could perhaps use the OLI data to generate a land mask, e.g. by a threshold on one of the SWIR bands

**Comment by jhonnatan2178** [comment]
But is that possible to do it in the same software, or do I need to do it in another software first? If the last one is true how could I incorporate the files then to Acolite?
